src/vm/types.rs

```diff
@@ -725,19 +730,25 @@ impl TypeSignature {
         // TODO: empty list type should be typed/handled differently.
         //         any list type should _admit_
         //         an empty list type
-        TypeSignature { atomic_type: AtomTypeIdentifier::IntType,
-                        list_dimensions: Some(ListTypeData { max_len: 0,
-                                                             dimension: 1 })}
+        TypeSignature::List(AtomTypeIdentifier::NoType,
+                            ListTypeData { max_len: 0,
+                                           dimension: 1 })
     }
 
     pub fn size(&self) -> Result<i128> {
-        let list_multiplier = match self.list_dimensions {
-                Some(ref list_data) => (list_data.max_len as i128).checked_mul(list_data.dimension as i128)
-                .ok_or(Error::new(ErrType::ValueTooLarge)),
-                None => Ok(1)
-        }?;
-        list_multiplier.checked_mul(self.atomic_type.size()?)
-            .ok_or(Error::new(ErrType::ValueTooLarge))
+        match self {
+            TypeSignature::List(ref atomic_type, ref list_data) => {
+                if list_data.max_len <= 0 {
+                    Ok(32 as i128)
+                } else {
+                    let multiplier = (list_data.max_len as i128).checked_mul(list_data.dimension as i128)
```

Comment:
Is it tractable to add a test on the error paths here?

Reply:
It might be, I think the computer would probably run out of memory first in these cases.

Reply:
Going with "intractable" then ;)